coalib/output/ConsoleInteraction.py

```diff
@@ -922,27 +933,49 @@ def print_bears(bears,
     :param show_params:      True if the parameters and their description
                              should be shown.
     :param console_printer:  Object to print messages on the console.
+    :param args:             Args passed to coala command.
     """
     if not bears:
         console_printer.print('No bears to show. Did you forget to install '
                               'the `coala-bears` package? Try `pip3 install '
                               'coala-bears`.')
         return
 
-    for bear, sections in sorted(bears.items(),
-                                 key=lambda bear_tuple:
-                                 bear_tuple[0].name.lower()):
-        show_bear(bear,
-                  show_description,
-                  show_params,
-                  console_printer)
+    results = [bear for bear, _ in sorted(bears.items(),
+                                          key=lambda bear_tuple:
+                                          bear_tuple[0].name.lower())]
+    if args and args.json:
+        from coalib.output.JSONEncoder import create_json_encoder
+        JSONEncoder = create_json_encoder(use_relpath=args.relpath)
+        json_output = {'bears': results}
+        import json
+        json_formatted_output = json.dumps(json_output,
+                                           cls=JSONEncoder,
+                                           sort_keys=True,
+                                           indent=2,
+                                           separators=(',', ': '))
+        if args.output:
+            filename = str(args.output[0])
+            with open(filename, 'w+') as fp:
```

Comment:
`w+`, so write and append? That would corrupt existing json files, the parent element has to be either an item or an array in json syntax.

Reply:
It is same as https://github.com/coala/coala/pull/4726/commits/a581d099b1d8b8dab3a910f3ef2e2a974e67338f#diff-a297897b536e74044ca36e47697b169dL84. @tulphoon has made last changes there. Should I add new commit to correct it at both the locations ?

Reply:
yeah sounds good :+1: